src/Workspaces/Remote/ServiceHub/Services/RemoteHostService.cs

```diff
@@ -65,12 +66,17 @@ public RemoteHostService(Stream stream, IServiceProvider serviceProvider)
             StartService();
         }
 
+        /// <summary>
+        /// Remote API.
+        /// </summary>
         public string Connect(string host, int uiCultureLCID, int cultureLCID, string? serializedSession, CancellationToken cancellationToken)
         {
             return RunService(() =>
             {
                 cancellationToken.ThrowIfCancellationRequested();
 
+                AssetStorage.SetAssetSource(this);
```

Comment:
could you comment this (and the one above).  it woul help me get a sense for the lifetimes/ownership.

Reply:
Sure. The storage is global (static).